include/eigenpy/details.hpp

```diff
@@ -117,8 +130,12 @@ namespace eigenpy
     bp::object NumpyMatrixObject; PyTypeObject * NumpyMatrixType;
     //bp::object NumpyAsMatrixObject; PyTypeObject * NumpyAsMatrixType;
     bp::object NumpyArrayObject; PyTypeObject * NumpyArrayType;
+    
+    static NP_TYPE np_type;
```

Comment:
I'm not deep into the code. But storing the type (array/matrix) statically doesn't seem to make sense for me on first sight.
Can we still have multiple `ndarray` instances with different type then?

Reply:
In fact, with EigenPy, you can return an Eigen::Matrix as either numpy.Matrix or numpy.Array depending on the convention you want to choose. By default, we use the notion of Matrix.

Reply:
I'm not sure that this should be a _global_ flag.

Reply:
The class is a Singleton, so it can be.